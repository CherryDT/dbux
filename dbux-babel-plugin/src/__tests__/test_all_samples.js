import path from 'path';

//import { promises as fs } from 'fs';
import fs from 'fs';
import { runSnapshotTests } from '../testing/test-util';

const codeFolder = path.join(__dirname, '__code__');

fs.readdirSync(codeFolder).forEach(fname => {
  //const fname = path.basename(fpath);
  const fpath = path.join(codeFolder, fname);
  const code = fs.readFileSync(fpath, { encoding: 'utf8' });

  expect(code).toBeString();
  
  runSnapshotTests(code, fpath, fname);
});