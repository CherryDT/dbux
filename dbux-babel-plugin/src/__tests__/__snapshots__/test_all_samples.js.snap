// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dbux-babel-plugin allFunctionTypes.es5.js: allFunctionTypes.es5.js 1`] = `
"
/* #################################################################################### */
function f1() {

}

const f2 = function _f2() {

};

const f3 = () => {

};

const o = {
  f4() {},
  f5: () => {}
};

class C {
  f6() {}
  f7 = () => {};
}

// // generator function
// function *f8() {
//   yield 1;
//   yield 2;
// }

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _dbuxRuntime = require('dbux-runtime')[\\"default\\"];

var _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  var f1 = function f1() {
    var _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  };

  var f2 = function _f2() {
    var _contextId2 = _dbux.pushImmediate(3);

    try {} finally {
      _dbux.popImmediate(_contextId2);
    }
  };

  var f3 = function f3() {
    var _contextId3 = _dbux.pushImmediate(4);

    try {} finally {
      _dbux.popImmediate(_contextId3);
    }
  };

  var o = {
    f4: function f4() {
      var _contextId4 = _dbux.pushImmediate(5);

      try {} finally {
        _dbux.popImmediate(_contextId4);
      }
    },
    f5: function f5() {
      var _contextId5 = _dbux.pushImmediate(6);

      try {} finally {
        _dbux.popImmediate(_contextId5);
      }
    }
  };

  var C =
  /*#__PURE__*/
  function () {
    function C() {
      this.f7 = function () {
        var _contextId6 = _dbux.pushImmediate(7);

        try {} finally {
          _dbux.popImmediate(_contextId6);
        }
      };
    }

    var _proto = C.prototype;

    _proto.f6 = function f6() {
      var _contextId7 = _dbux.pushImmediate(8);

      try {} finally {
        _dbux.popImmediate(_contextId7);
      }
    };

    return C;
  }(); // // generator function
  // function *f8() {
  //   yield 1;
  //   yield 2;
  // }

} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"allFunctionTypes.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/allFunctionTypes.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"_f2\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"line\\": 10,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"name\\": \\"f4\\",
      \\"line\\": 15,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 6,
      \\"type\\": 2,
      \\"name\\": \\"f5\\",
      \\"line\\": 16,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 7,
      \\"type\\": 2,
      \\"line\\": 21,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 8,
      \\"type\\": 2,
      \\"name\\": \\"f6\\",
      \\"line\\": 20,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin allFunctionTypes.esNext.js: allFunctionTypes.esNext.js 1`] = `
"
/* #################################################################################### */
function f1() {

}

const f2 = function _f2() {

};

const f3 = () => {

};

const o = {
  f4() {},
  f5: () => {}
};

class C {
  f6() {}
  f7 = () => {};
}

// // generator function
// function *f8() {
//   yield 1;
//   yield 2;
// }

      ↓ ↓ ↓ ↓ ↓ ↓

const _dbuxRuntime = require('dbux-runtime').default;

const _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  function f1() {
    const _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  }

  const f2 = function _f2() {
    const _contextId2 = _dbux.pushImmediate(3);

    try {} finally {
      _dbux.popImmediate(_contextId2);
    }
  };

  const f3 = () => {
    const _contextId3 = _dbux.pushImmediate(4);

    try {} finally {
      _dbux.popImmediate(_contextId3);
    }
  };

  const o = {
    f4() {
      const _contextId4 = _dbux.pushImmediate(5);

      try {} finally {
        _dbux.popImmediate(_contextId4);
      }
    },

    f5: () => {
      const _contextId5 = _dbux.pushImmediate(6);

      try {} finally {
        _dbux.popImmediate(_contextId5);
      }
    }
  };

  class C {
    constructor() {
      this.f7 = () => {
        const _contextId6 = _dbux.pushImmediate(7);

        try {} finally {
          _dbux.popImmediate(_contextId6);
        }
      };
    }

    f6() {
      const _contextId7 = _dbux.pushImmediate(8);

      try {} finally {
        _dbux.popImmediate(_contextId7);
      }
    }

  } // // generator function
  // function *f8() {
  //   yield 1;
  //   yield 2;
  // }

} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"allFunctionTypes.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/allFunctionTypes.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"f1\\",
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"_f2\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"line\\": 10,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"name\\": \\"f4\\",
      \\"line\\": 15,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 6,
      \\"type\\": 2,
      \\"line\\": 16,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 7,
      \\"type\\": 2,
      \\"line\\": 21,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 8,
      \\"type\\": 2,
      \\"name\\": \\"f6\\",
      \\"line\\": 20,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin example1.es5.js: example1.es5.js 1`] = `
"
/* #################################################################################### */
function f1(x,y) {
}

class A {
  constructor() {
    this.m = new Map();
    this.s = new Set();
  }

  getX() {}
}

const f2 = (z,w) => {};

export {
  f1,
  A,
  f2
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

require(\\"core-js/modules/es.array.iterator\\");

require(\\"core-js/modules/es.map\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.set\\");

require(\\"core-js/modules/es.string.iterator\\");

require(\\"core-js/modules/web.dom-collections.iterator\\");

exports.__esModule = true;
exports.f2 = exports.A = exports.f1 = void 0;

var _dbuxRuntime = require('dbux-runtime')[\\"default\\"];

var _dbux = _dbux_init(_dbuxRuntime);

var _f, _A, _f2;

exports.f2 = _f2;
exports.A = _A;
exports.f1 = _f;

try {
  /* #################################################################################### */
  var f1 = function f1(x, y) {
    var _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  };

  var A =
  /*#__PURE__*/
  function () {
    function A() {
      var _contextId2 = _dbux.pushImmediate(3);

      try {
        this.m = new Map();
        this.s = new Set();
      } finally {
        _dbux.popImmediate(_contextId2);
      }
    }

    var _proto = A.prototype;

    _proto.getX = function getX() {
      var _contextId3 = _dbux.pushImmediate(4);

      try {} finally {
        _dbux.popImmediate(_contextId3);
      }
    };

    return A;
  }();

  var f2 = function f2(z, w) {
    var _contextId4 = _dbux.pushImmediate(5);

    try {} finally {
      _dbux.popImmediate(_contextId4);
    }
  };

  exports.f1 = _f = f1;
  exports.A = _A = A;
  exports.f2 = _f2 = f2;
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"example1.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/example1.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"A\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"name\\": \\"getX\\",
      \\"line\\": 11,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"line\\": 14,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin example1.esNext.js: example1.esNext.js 1`] = `
"
/* #################################################################################### */
function f1(x,y) {
}

class A {
  constructor() {
    this.m = new Map();
    this.s = new Set();
  }

  getX() {}
}

const f2 = (z,w) => {};

export {
  f1,
  A,
  f2
}

      ↓ ↓ ↓ ↓ ↓ ↓

const _dbuxRuntime = require('dbux-runtime').default;

const _dbux = _dbux_init(_dbuxRuntime);

let _f, _A, _f2;

try {
  /* #################################################################################### */
  function f1(x, y) {
    const _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  }

  class A {
    constructor() {
      const _contextId2 = _dbux.pushImmediate(3);

      try {
        this.m = new Map();
        this.s = new Set();
      } finally {
        _dbux.popImmediate(_contextId2);
      }
    }

    getX() {
      const _contextId3 = _dbux.pushImmediate(4);

      try {} finally {
        _dbux.popImmediate(_contextId3);
      }
    }

  }

  const f2 = (z, w) => {
    const _contextId4 = _dbux.pushImmediate(5);

    try {} finally {
      _dbux.popImmediate(_contextId4);
    }
  };

  _f = f1;
  _A = A;
  _f2 = f2;
} finally {
  _dbux.popProgram();
}

export { _f as f1, _A as A, _f2 as f2 };

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"example1.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/example1.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"f1\\",
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"constructor\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"name\\": \\"getX\\",
      \\"line\\": 11,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"line\\": 14,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin example2.es5.js: example2.es5.js 1`] = `
"
/* #################################################################################### */
import { A } from './example1';

(function main() {
  console.log(new A());
})();

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _example = require(\\"./example1\\");

/* #################################################################################### */
var _dbuxRuntime = require('dbux-runtime')[\\"default\\"];

var _dbux = _dbux_init(_dbuxRuntime);

try {
  (function main() {
    var _contextId = _dbux.pushImmediate(2);

    try {
      console.log(new _example.A());
    } finally {
      _dbux.popImmediate(_contextId);
    }
  })();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"example2.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/example2.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"main\\",
      \\"line\\": 4,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin example2.esNext.js: example2.esNext.js 1`] = `
"
/* #################################################################################### */
import { A } from './example1';

(function main() {
  console.log(new A());
})();

      ↓ ↓ ↓ ↓ ↓ ↓

/* #################################################################################### */
import { A } from './example1';

const _dbuxRuntime = require('dbux-runtime').default;

const _dbux = _dbux_init(_dbuxRuntime);

try {
  (function main() {
    const _contextId = _dbux.pushImmediate(2);

    try {
      console.log(new A());
    } finally {
      _dbux.popImmediate(_contextId);
    }
  })();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"example2.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/example2.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"main\\",
      \\"line\\": 4,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin executionContextAsync.es5.js: executionContextAsync.es5.js 1`] = `
"
/* #################################################################################### */
async function af1() {
  return Promise.resolve(1);
}

async function af2() {
  await af1();
  await af1();

  return 2 + await af1();
}

async function af33(delay) {
  return new Promise(r => {
    setTimeout(r.bind(33), delay);
  });
}

async function af44() {
  const a = await af33(100);
  const b = await af33(50);

  return a+b + af33(10);
}

(async function main() {
  return Promise.all([
    af2(),
    af44()
  ]);
})();

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

require(\\"core-js/modules/es.array.iterator\\");

require(\\"core-js/modules/es.function.bind\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.promise\\");

require(\\"core-js/modules/es.string.iterator\\");

require(\\"core-js/modules/web.dom-collections.iterator\\");

require(\\"core-js/modules/web.timers\\");

require(\\"regenerator-runtime/runtime\\");

var _dbuxRuntime = require('dbux-runtime')[\\"default\\"];

var _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  var af1 = function af1() {
    var _contextId;

    return regeneratorRuntime.async(function af1$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _contextId = _dbux.pushImmediate(2);
            _context.prev = 1;
            return _context.abrupt(\\"return\\", Promise.resolve(1));

          case 3:
            _context.prev = 3;

            _dbux.popImmediate(_contextId);

            return _context.finish(3);

          case 6:
          case \\"end\\":
            return _context.stop();
        }
      }
    }, null, null, [[1,, 3, 6]]);
  };

  var af2 = function af2() {
    var _contextId2;

    return regeneratorRuntime.async(function af2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            _contextId2 = _dbux.pushImmediate(3);
            _context2.prev = 1;
            _context2.next = 4;
            return regeneratorRuntime.awrap(af1());

          case 4:
            _context2.next = 6;
            return regeneratorRuntime.awrap(af1());

          case 6:
            _context2.next = 8;
            return regeneratorRuntime.awrap(af1());

          case 8:
            _context2.t0 = _context2.sent;
            return _context2.abrupt(\\"return\\", 2 + _context2.t0);

          case 10:
            _context2.prev = 10;

            _dbux.popImmediate(_contextId2);

            return _context2.finish(10);

          case 13:
          case \\"end\\":
            return _context2.stop();
        }
      }
    }, null, null, [[1,, 10, 13]]);
  };

  var af33 = function af33(delay) {
    var _contextId3;

    return regeneratorRuntime.async(function af33$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            _contextId3 = _dbux.pushImmediate(4);
            _context3.prev = 1;
            return _context3.abrupt(\\"return\\", new Promise(function (r) {
              var _contextId5 = _dbux.pushImmediate(6);

              try {
                var _contextId4 = _dbux.pushImmediate(5);

                try {
                  setTimeout(r.bind(33), delay);
                } finally {
                  _dbux.popImmediate(_contextId4);
                }
              } finally {
                _dbux.popImmediate(_contextId5);
              }
            }));

          case 3:
            _context3.prev = 3;

            _dbux.popImmediate(_contextId3);

            return _context3.finish(3);

          case 6:
          case \\"end\\":
            return _context3.stop();
        }
      }
    }, null, null, [[1,, 3, 6]]);
  };

  var af44 = function af44() {
    var _contextId6, _a, _b;

    return regeneratorRuntime.async(function af44$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            _contextId6 = _dbux.pushImmediate(7);
            _context4.prev = 1;
            _context4.next = 4;
            return regeneratorRuntime.awrap(af33(100));

          case 4:
            _a = _context4.sent;
            _context4.next = 7;
            return regeneratorRuntime.awrap(af33(50));

          case 7:
            _b = _context4.sent;
            return _context4.abrupt(\\"return\\", _a + _b + af33(10));

          case 9:
            _context4.prev = 9;

            _dbux.popImmediate(_contextId6);

            return _context4.finish(9);

          case 12:
          case \\"end\\":
            return _context4.stop();
        }
      }
    }, null, null, [[1,, 9, 12]]);
  };

  (function main() {
    var _contextId7;

    return regeneratorRuntime.async(function main$(_context5) {
      while (1) {
        switch (_context5.prev = _context5.next) {
          case 0:
            _contextId7 = _dbux.pushImmediate(8);
            _context5.prev = 1;
            return _context5.abrupt(\\"return\\", Promise.all([af2(), af44()]));

          case 3:
            _context5.prev = 3;

            _dbux.popImmediate(_contextId7);

            return _context5.finish(3);

          case 6:
          case \\"end\\":
            return _context5.stop();
        }
      }
    }, null, null, [[1,, 3, 6]]);
  })();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"executionContextAsync.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/executionContextAsync.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"line\\": 13,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"line\\": 14,
      \\"parent\\": 4
    }, {
      \\"staticId\\": 6,
      \\"type\\": 2,
      \\"line\\": 14,
      \\"parent\\": 4
    }, {
      \\"staticId\\": 7,
      \\"type\\": 2,
      \\"line\\": 19,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 8,
      \\"type\\": 2,
      \\"name\\": \\"main\\",
      \\"line\\": 26,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin executionContextAsync.esNext.js: executionContextAsync.esNext.js 1`] = `
"
/* #################################################################################### */
async function af1() {
  return Promise.resolve(1);
}

async function af2() {
  await af1();
  await af1();

  return 2 + await af1();
}

async function af33(delay) {
  return new Promise(r => {
    setTimeout(r.bind(33), delay);
  });
}

async function af44() {
  const a = await af33(100);
  const b = await af33(50);

  return a+b + af33(10);
}

(async function main() {
  return Promise.all([
    af2(),
    af44()
  ]);
})();

      ↓ ↓ ↓ ↓ ↓ ↓

const _dbuxRuntime = require('dbux-runtime').default;

const _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  async function af1() {
    const _contextId = _dbux.pushImmediate(2);

    try {
      return Promise.resolve(1);
    } finally {
      _dbux.popImmediate(_contextId);
    }
  }

  async function af2() {
    const _contextId2 = _dbux.pushImmediate(3);

    try {
      await af1();
      await af1();
      return 2 + (await af1());
    } finally {
      _dbux.popImmediate(_contextId2);
    }
  }

  async function af33(delay) {
    const _contextId3 = _dbux.pushImmediate(4);

    try {
      return new Promise(r => {
        const _contextId4 = _dbux.pushImmediate(5);

        try {
          setTimeout(r.bind(33), delay);
        } finally {
          _dbux.popImmediate(_contextId4);
        }
      });
    } finally {
      _dbux.popImmediate(_contextId3);
    }
  }

  async function af44() {
    const _contextId5 = _dbux.pushImmediate(6);

    try {
      const a = await af33(100);
      const b = await af33(50);
      return a + b + af33(10);
    } finally {
      _dbux.popImmediate(_contextId5);
    }
  }

  (async function main() {
    const _contextId6 = _dbux.pushImmediate(7);

    try {
      return Promise.all([af2(), af44()]);
    } finally {
      _dbux.popImmediate(_contextId6);
    }
  })();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"executionContextAsync.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/executionContextAsync.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"af1\\",
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"af2\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 4,
      \\"type\\": 2,
      \\"name\\": \\"af33\\",
      \\"line\\": 13,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 5,
      \\"type\\": 2,
      \\"line\\": 14,
      \\"parent\\": 4
    }, {
      \\"staticId\\": 6,
      \\"type\\": 2,
      \\"name\\": \\"af44\\",
      \\"line\\": 19,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 7,
      \\"type\\": 2,
      \\"name\\": \\"main\\",
      \\"line\\": 26,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin executionContextSimple.es5.js: executionContextSimple.es5.js 1`] = `
"
/* #################################################################################### */
function f1() {

}

function f2() {
  f1();
  f1();
}


f2();

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _dbuxRuntime = require('dbux-runtime')[\\"default\\"];

var _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  var f1 = function f1() {
    var _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  };

  var f2 = function f2() {
    var _contextId2 = _dbux.pushImmediate(3);

    try {
      f1();
      f1();
    } finally {
      _dbux.popImmediate(_contextId2);
    }
  };

  f2();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"executionContextSimple.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/executionContextSimple.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"line\\": 6,
      \\"parent\\": 1
    }]
  });
}
"
`;

exports[`dbux-babel-plugin executionContextSimple.esNext.js: executionContextSimple.esNext.js 1`] = `
"
/* #################################################################################### */
function f1() {

}

function f2() {
  f1();
  f1();
}


f2();

      ↓ ↓ ↓ ↓ ↓ ↓

const _dbuxRuntime = require('dbux-runtime').default;

const _dbux = _dbux_init(_dbuxRuntime);

try {
  /* #################################################################################### */
  function f1() {
    const _contextId = _dbux.pushImmediate(2);

    try {} finally {
      _dbux.popImmediate(_contextId);
    }
  }

  function f2() {
    const _contextId2 = _dbux.pushImmediate(3);

    try {
      f1();
      f1();
    } finally {
      _dbux.popImmediate(_contextId2);
    }
  }

  f2();
} finally {
  _dbux.popProgram();
}

function _dbux_init() {
  return _dbuxRuntime.initProgram({
    \\"fileName\\": \\"executionContextSimple.js\\",
    \\"filePath\\": \\"samples/__samplesInput__/executionContextSimple.js\\",
    \\"staticSites\\": [null, {
      \\"staticId\\": 1,
      \\"type\\": 1,
      \\"name\\": \\"Program\\"
    }, {
      \\"staticId\\": 2,
      \\"type\\": 2,
      \\"name\\": \\"f1\\",
      \\"line\\": 2,
      \\"parent\\": 1
    }, {
      \\"staticId\\": 3,
      \\"type\\": 2,
      \\"name\\": \\"f2\\",
      \\"line\\": 6,
      \\"parent\\": 1
    }]
  });
}
"
`;
